web/app/view/ArrayListFilter.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2017 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+Ext.define('Traccar.view.ArrayListFilter', {
+    extend: 'Ext.grid.filters.filter.List',
+    alias: 'grid.filter.arraylist',
+
+    type: 'arraylist',
+
+    constructor: function (config) {
+        var me = this;
+        me.callParent([config]);
+        me.filter.setFilterFn(function (item) {
```

Comment:
Grid filters is UI above store filters. List filter perfectly fits to us, except he can not compare arrays.

Store filters can be set in two ways:
- `value` and `operator` http://docs.sencha.com/extjs/6.2.0/classic/Ext.util.Filter.html#cfg-value
http://docs.sencha.com/extjs/6.2.0/classic/Ext.util.Filter.html#cfg-operator 
Every store item (its field) compared with `value` with help of `operator`. List uses 'in' operator.
- custom `filterFn` function http://docs.sencha.com/extjs/6.2.0/classic/Ext.util.Filter.html#cfg-filterFn

`filterFn` has priority.

There is no default operator to intersect arrays, http://docs.sencha.com/extjs/6.2.0/classic/Ext.util.Filter.html#cfg-operator

because of that we should implement custom `filterFn` function that should return true if Device visible.
At least one `geofenceIds` array item contains in `value` array.

Now about function itself:

`this.getProperty()` return `dataIndex` value of the column. It can be replaced to `'geofenceIds'` directly, but component won't be universal.
`property` variable will be `device.geofenceIds` of every device
`value` is the value generated by `filter.List`  basically what user selected in dropdown menu.
Next we just look every item from one array in other.
Or if `property` not array just look in array.